### ISSUE TITLE
gethostbyname logic + example issues

### ISSUE THREAD

**Issue by dbough** [HelgeSverre/Domain-Availability#9]
Hi -

On ~line 418 in `AvailabilityScript.php` there is a check to determine if the domain name == the ipv4 address (as returned by `gethostbyname`).  This will always fail meaning `isAvaible` will always return false.  Not sure why this is being done?:

    if (gethostbyname($domain) == $domain) { 
        $tld = $this->getTld($domain);

Also, the logic in the README for the example usage is wrong. This:

    if ($available) { 
        echo $domain." is not registered\n";
    } else {
        echo $domain." is registered\n";
    }

Should probably be something like this:

    if ($available === false) {
        echo $domain." is not registered\n";
    } else {
        echo $domain." is registered\n";
    }

**Comment by dbough** [comment]
I changed this:

    if (gethostbyname($domain) == $domain) { 
        $tld = $this->getTld($domain);

To this:

    if (gethostbyname($domain)) { 
        $tld = $this->getTld($domain);

After which I found another issue a bit lower in the logic.  This is saying the only way we're returning true is if the $badString exists:

                if (strpos($whois, $badString) !== false) {
                    return true; // available
                }
                return false; // not available

It should be something like this:

                if (strpos($whois, $badString) === false) {
                    return true; // available
                }
                return false; // not available

**Comment by HelgeSverre** [comment]
I'll go through the entire script and refactor it in a little while.

**Comment by mirr** [comment]
I was about to create a new issue but this solved my problem. The script was always saying the domain is registered even though domain is available.

**Comment by vitushoesl** [comment]
Had exactly the same problem, but the solution is not correct (if I understood it correctly). In the script I downloaded right now the original code is:

```
                if (strpos($whois, $badString) === false) {
					return true; // available
				}
				return false; // not available
```

That's wrong. It does exactly the wrong thing. If the `$badString` doesn't appear in the whois respond it says, the domain is available (because `strpos($whois, $badString)` returns false when the snipped doesn't appear).

You have to use
```
                if (strpos($whois, $badString) !== false) {
					return true; // available
				}
	            return false; // not available
```

If the string doesn't appear, `strpos($whois, $badString)` is the number of the position where the `$badString` appears. In this case, the domain is available. In all other cases this will be false, because thats what `strpos($whois, $badString)` returns when the string is not correct.

Changed your script this way and now it works great.

**Comment by vitushoesl** [comment]
Another thing I've got trouble with what that - at least in my case - `gethostbyname()` returns the ip adress of itself if the domain is not available. I changed the first if condition (l.419) from

```
if (gethostbyname($domain) == $domain) {
```

to

```
if (gethostbyname($domain) == gethostbyname(gethostname())) {
```

If the domain is in use, it usally returns the ip adress of the dns records. If it returns my own ip adress all the process with the whois information is done.

Works for me as well.
